src/DataStructures/ModalVector.hpp

```diff
@@ -0,0 +1,108 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/PointerVector.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_include <blaze/math/expressions/DVecMapExpr.h>
+// IWYU pragma: no_include <blaze/math/typetraits/IsVector.h>
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A class for storing spectral coefficients on a mesh.
+ *
+ * \details A ModalVector holds an array of spectral coefficients, and can be
+ * either owning (the array is deleted when the ModalVector goes out of scope)
+ * or non-owning, meaning it just has a pointer to an array.
+ *
+ * Only basic mathematical operations are supported with
+ * ModalVectors. ModalVectors may be added or subtracted, and the following
+ * unary operations are supported:
+ * - max
+ * - min
```

Comment:
Do we want to support `abs`? For AMR operations it is quite common to look at the absolute value of the coefficients.

Reply:
from the old conversations on #704, @kidder was questioning whether the absolute value of a `ModalVector` should be another `ModalVector`. Though, perhaps the practicality of wanting to numerically compute several absolute values quickly should rule in favor of permitting the operation.

I'll go ahead and add it; I just want to confirm that all of the type checking is as we intended.

Reply:
I talked with @kidder in person and we agreed that the result should be a `ModalVector` :)